### ISSUE TITLE
creditcard validation throws error only on mobile Safari

### ISSUE THREAD

**Issue by bradmurray** [jquery-validation/jquery-validation#1767]
Versions:
- jquery 2.2.3
- jquery-validate version 1.15.0
- mobile Safari on iOS 9.3 (confirmed on both iPhone 5 and iPhone 6s+

This code works fine on all other tested browsers (desktop and mobile) but in mobile Safari when calling the .validate for the page I receive:

```
[Error] TypeError: undefined is not an object (evaluating 'a.validator.methods[d].call').  Exception occurred when checking element txtCreditCardNumber, check the 'creditcard' method.
    check (jquery.validate.min.js:4:11271)
    element (jquery.validate.min.js:4:7203)
    onfocusout (jquery.validate.min.js:4:3486)
    b (jquery.validate.min.js:4:5194)
    dispatch (jquery-2.2.3.min.js:3:7543)
    handle (jquery-2.2.3.min.js:3:5626)
    trigger (jquery-2.2.3.min.js:4:4842)
    simulate (jquery-2.2.3.min.js:4:5262)
    c (jquery-2.2.3.min.js:4:6025)
```

the error is reported as coming from jquery.validate.min.js:276

The rule for this field is simply:

``` JSON
{ required: true, creditcard: true }
```

My current workaround is to change the rule to:

``` JSON
{ required: true, digits: true, minlength: 12, maxlength: 16 }
```

when I detect an iOS device. This seems to work but is not ideal.

**Comment by staabm** [comment]
Could you provide a reduced jsfiddle which reproduces your issue?

**Comment by bradmurray** [comment]
I distilled it down as best as I could to this: https://jsfiddle.net/xt6yasLt/14/
But I can't seem to get it to throw the same problem in the emulator with this code.

**Comment by pirco** [comment]
having the same issue. not only safari iOS but safari desktop (v12 on OS mojave)

**Comment by Arkni** [comment]
Hi all,

Re-opening this issue as the bug still persist. I will try to fix the issue as soon as I got some free time, but if anyone is interested in fixing this issue and submit a proposal, please do. I'd be more than happy to walk you through the steps involved.

And, sorry about the closing of this issue by our bot.

**Comment by carlos-mg89** [comment]
Something similar is happening to me, in Firefox 72.0.1 on Ubuntu 19.10 but with the minLength rule.

`Exception occurred when checking element password, check the 'minLength' method. TypeError: "a.validator.methods[d] is undefined"`

The error seems to happen only for fields with the minLength or maxLength restrictions.

This seems to be quite an important bug, to be honest. I'd be happy to try to fix this bug. Any guidance would be more than welcome @Arkni

**Comment by carlos-mg89** [comment]
I've tried with jQuery 3.1 and 3.4 with no luck. Perhaps that's the reason of the bug.

I'm using the jQuery migrate as well but it doesn't improve.

**Comment by carlos-mg89** [comment]
jQuery Core 2.2.4 doesn't help wither.

**Comment by carlos-mg89** [comment]
Alright, I've already figured out what was going on. This plugins works great with jQuery 3.4.1 and without the jQuery migration script. 

Except when you use the minLength and maxLength restrictions. Then I saw issues when using the highlight / unhighlight JS options. If not using them, then it works normally.
